File: Practica01/EcuacionLineal.java
import java.util.*;
public class EcuacionLineal{
    private double a,b,c,d,e,f;
    public EcuacionLineal(double a,double b,double c,double d,double e,double f) {
        this.a=a;
        this.b=b;
        this.c=c;
        this.d=d;
        this.e=e;
        this.f=f;
    }
    public boolean tieneSolucion() {
        return a*d-b*c!=0;
    }
    public double getX() {
        return (e*d-b*f)/(a*d-b*c);
    }
    public double getY() {
        return (a*f-e*c)/(a*d-b*c);
    }
    public static void main(String[] args) {
        Scanner leer=new Scanner(System.in);
        double a,b,c,d,e,f;
        System.out.print("Ingrese a: ");
        a=leer.nextDouble();
        System.out.print("Ingrese b: ");
        b=leer.nextDouble();
        System.out.print("Ingrese c: ");
        c=leer.nextDouble();
        System.out.print("Ingrese d: ");
        d=leer.nextDouble();
        System.out.print("Ingrese e: ");
        e=leer.nextDouble();
        System.out.print("Ingrese f: ");
        f=leer.nextDouble();
        EcuacionLineal ecuacion=new EcuacionLineal(a, b, c, d, e, f);
        if (ecuacion.tieneSolucion()) {
            System.out.println("x es: "+ecuacion.getX());
            System.out.println("y es: "+ecuacion.getY());
        }else {
            System.out.println("No hay solucion");
        }
    } 
    }

File: Practica01/EcuacionLineal.py
class EcuacionLineal:
    def __init__(self,a,b,c,d,e,f):
        self.__a=a
        self.__b=b
        self.__c=c
        self.__d=d
        self.__e=e
        self.__f=f
    def tieneSolucion(self):
        return self.__a*self.__d -self.__b*self.__c!=0
    def getX(self):
        return (self.__e*self.__d-self.__b*self.__f)/(self.__a*self.__d-self.__b*self.__c)
    def getY(self):
        return (self.__a*self.__f-self.__e*self.__c)/(self.__a*self.__d-self.__b*self.__c)
a=float(input("Ingrese a: "))
b=float(input("Ingrese b: "))
c=float(input("Ingrese c: "))
d=float(input("Ingrese d: "))
e=float(input("Ingrese e: "))
f=float(input("Ingrese f: "))
ecuacion=EcuacionLineal(a,b,c,d,e,f)
if ecuacion.tieneSolucion():
    print("x es: ",ecuacion.getX())
    print("y es: ",ecuacion.getY())
else:
    print("No hay solucion")
    
